ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -683,20 +681,17 @@ Operator genOPTree(ASTNode ast, PlannerContext plannerCtx) throws SemanticExcept
             // Wrap all other errors (Should only hit in tests)
             throw new SemanticException(e);
           } else {
-            reAnalyzeAST = true;
+            String strategies = conf.getVar(ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES);
+            if (strategies == null || !Arrays.stream(strategies.split(",")).anyMatch("recompile_without_cbo"::equals)) {
+              throw new SemanticException("Invalid configuration. If fallbackStrategy is set to " + fallbackStrategy.name() +
+                  " then " + ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES.varname + " should contain 'recompile_without_cbo'");
```

Comment:
if this is a sanity check then I think this exception should be thrown all the time regardless we are trying to fall back or not.

Imagine the following:
almost all queries are handled by CBO; and then comes 1 query and they hit this exception....instead of seeing what the issue is they will start investigating this exception....

you could move this check somewhere into `ReCompilePlugin` - and call that this check when the plugin is not initialized -  I think that would be a better place to throw this exception; or not throw it at all.

...or we could add the `ReCompilePlugin` based on the actual setting of the `fallbackStrategy`; but in that case this will became a secret contract...(there's always something on the flipside :D)
...but automating things is also problematic; because if we add it automatically shouldn't we also throw an exception if the reexecution is disabled alltogether?

Reply:
Moved the check to `ReExecDriver.checkHookConfig` and throwing an exception whenever the config is not correct